### PR TITLE
Upgrade to the latest PHP Markdown Lib for PHP 7 compatibility.

### PR DESCRIPTION
**Pull request by frjo**
Just committed this to the Drupal version, see https://www.drupal.org/node/2766143.

### CODE REVIEW

File: markdown.module [hunk 1]
```diff
@@ -103,8 +103,8 @@ And now some code:
  */
 function _filter_markdown($text, $format) {
   if (!empty($text)) {
-    module_load_include('php', 'markdown', 'markdown');
-    $text = Markdown($text);
+    module_load_include('inc.php', 'markdown', 'includes/MarkdownExtra');
```
Comment (deekayen) [inline markdown.module:106]: I don't think switching from Markdown to MarkdownExtra is a trivial change, that's why I made the Markdown Extra module: https://github.com/backdrop-contrib/markdownextra

File: markdown.module [hunk 2]
```diff
@@ -114,15 +114,14 @@ function _filter_markdown($text, $format) {
  * Filter settings callback. Just provides a version overview.
  */
 function _filter_markdown_settings($form, &$form_state, $filter, $format) {
-  module_load_include('php', 'markdown', 'markdown');
+  module_load_include('inc.php', 'markdown', 'includes/MarkdownExtra');
```
Comment (deekayen) [inline markdown.module:117]: Another non-trivial change to Markdown Extra.

**Comment by deekayen** [comment]
I don't mind deprecating/deleting the Markdown Extra project if you switch to Markdown Extra - I'm obviously in favor of using that variant, however it ought to come with a README update and a separate note in the changelog for the point release.

**Comment by frjo** [comment]
The Drupal version of this module have always used the Markdown Extra version and from what I can see the Backdrop version is no different. See https://github.com/backdrop-contrib/markdown/blob/1.x-1.x/markdown.php#L57

I see now that https://github.com/backdrop-contrib/markdownextra already uses the latest PHP Markdown lib so maybe just scrap this version or merge them.

**Comment by herbdool** [comment]
Given that the Drupal Markdown module is well-known and already includes Markdown Extra, it might be better to preserve this name and track that module as close as possible.

I noticed the Drupal's 7.x-2.x branch actually requires including the Markdown Extra library manually. But it has no official release.

**Comment by deekayen** [comment]
If this gets merged, I'll wipe my fork and point the readme back at this project. Still looking for @chrisherberte to weigh in.

**Comment by frjo** [comment]
The 7.x-2.x branch was an attempt to support multiple markdown libraries. The Drupal 8 version has that feature but 7.x-2.x is dead, would delete it if possible.

**Comment by herbdool** [comment]
@chrisherberte are you still interested in maintaining this module? If not perhaps @frjo is interested in being added as a maintainer and keeping this version in line with the Drupal version.

The policy around this is outlined here: https://github.com/backdrop-ops/contrib -- if Chris doesn't responded within a couple weeks then it's considered abandoned and someone else who is a Backdrop contributer can take it on.

**Comment by deekayen** [comment]
It's been three weeks without a response from @chrisherberte. Is this where I invoke the power of @quicksketch to merge PR #3?

**Pull request merged**